XLS-0066d-lending-protocol/README.md

```diff
@@ -0,0 +1,1428 @@
+```
+
+##### 3.2.5.3 Failure Conditions
+
+- A `Loan` object with specified `LoanID` does not exist on the ledger.
+
+- The Loan has not started yet: `Loan.StartDate > CurrentTime`.
+
+- The submitter `AccountRoot.Account` is not equal to `Loan.Borrower`.
+
+- `Loan.PaymentRemaining` or `Loan.PrincipalOutstanding` is `0`.
+
+- If the `Vault(LoanBroker(Loan(LoanID).LoanBrokerID).VaultID).Asset` is an `IOU`:
+
+  - The trustline between the submitter account and the `Issuer` of the asset is frozen.
+  - The `AccountRoot` object of the `Issuer` has the `lsfGlobalFreeze` flag set.
+
+- If the `Vault(LoanBroker(Loan(LoanID).LoanBrokerID).VaultID).Asset` is an `MPT`:
+
+  - The `MPToken` object for the `Vault(LoanBroker(Loan(LoanID).LoanBrokerID).VaultID).Asset` of the submitter `AccountRoot`:
+    - Has `lsfMPTLocked` flag set.
+  - The `MPTokenIssuance` object of the `Vault(LoanBroker(Loan(LoanID).LoanBrokerID).VaultID).Asset` has the `lsfMPTLocked` flag set.
+
+- If `CurrentTime > Loan.NextPaymentDueDate` and `Amount` < `LatePaymentAmount()`
+
+- If `CurrentTime <= Loan.NextPaymentDueDate` and `Amount` < `PeriodicPaymentAmount()`
+
+##### 3.2.5.4 State Changes
+
+Assume the payment is split into `principal`, `interest` and `fee`, and `totalDue = principal + interest + fee`.
+
+Assume the payment is handled by a function that implements the [Pseudo-Code](#3252-transaction-pseudo-code) that returns `principal_paid`, `interest_paid`, `value_change` and `fee_paid`, where:
+
+- `principal_paid` is the amount of principal that the payment covered.
+- `interest_paid` is the amount of interest that the payment covered.
+- `fee_paid` is the amount of fee that the payment covered.
+- `value_change` is the amount by which the total value of the Loan changed.
+  - If `value_change` < `0`, Loan value decreased.
+  - If `value_change` > `0`, Loan value increased.
+
+Furthermore, assume `full_periodic_payments` variable represents the number of payment intervals that the payment covered.
+
+- `Loan` object state changes:
+
+  - If `Loan(LoanID).Flags == lsfLoanImpaired`:
+
+    - `Loan(LoanID).Flags = 0`
+
+  - Decrease `Loan.PaymentRemaining` by `full_periodic_payments`.
+  - Decrease `Loan.PrincipalOutstanding` by `principal_paid`.
+
+  - If `Loan.PaymentRemaining > 0` and `LoanPrincipalOutstanding > 0`:
+
+    - Set the next payment date: `Loan.NextPaymentDueDate += Loan.PaymentInterval * full_periodic_payments`.
+    - Set the previous payment date: `Loan.PreviousPaymentDate = Loan.NextPaymentDueDate - Loan.PaymentInterval`.
+
+- `LoanBroker(Loan.LoanBrokerID)` object state changes:
+
+  - Compute the management fee:
+
+    - `feeManagement = interest_paid x LoanBroker.ManagementFeeRate`
+
+  - If there is **not enough** first-loss capital: `LoanBroker.CoverAvailable < LoanBroker.DebtTotal x LoanBroker.CoverRateMinimum`:
+
+    - Do not charge the management, add it to the total debt:
+      - `LoanBroker.DebtTotal += feeManagement`
```

Comment:
This is basically a repeat of a question I asked earlier, but why is `feeManagement`, which is supposedly going to the broker being _added_ to the broker's debt? Shouldn't it either _come off_ the debt total, or be added to the `CoverAvailable`?

Reply:
See my comment to the first question. tl;dr: good idea.